### PR TITLE
WINDUP-1277 replaced Bcel library and ASM4 to ASM5 constants

### PR DESCRIPTION
**Pull request by mareknovotny**
https://issues.jboss.org/browse/WINDUP-1277

### CODE REVIEW

File: rules-java/tests/src/test/java/org/jboss/windup/rules/apps/java/ClassMetadataTest.java [hunk 1]
```diff
@@ -111,6 +111,14 @@ private void runTest(String inputPath) throws Exception
...
             Assert.assertEquals(1, interfaceCountFound);
+
+            // long count = 0;
+            // for (JavaMethodModel javaMethodM : javaClassModel.getMethod("testMethodDefault"))
+            // {
+            // Assert.assertEquals("testMethodDefault", javaMethodM.getMethodName());
+            // count++;
+            // }
+            // Assert.assertEquals(1, count);
         }
```
Comment (mareknovotny) [inline rules-java/tests/src/test/java/org/jboss/windup/rules/apps/java/ClassMetadataTest.java]: Tried this extension of test, but it seems that we are not adding any methods into javaClassModel. Isn't that bug or am I missing something?

Comment (jsight) [inline rules-java/tests/src/test/java/org/jboss/windup/rules/apps/java/ClassMetadataTest.java]: I think that we intentionally do not log this data. It isn't used by the rules anywhere.

File: rules-java/tests/src/test/resources/classmetadatatest/sampleclasses/com/rhc/booking/services/EventServer.java [hunk 2]
```diff
@@ -1,8 +0,0 @@
-package com.rhc.booking.services;
```
Comment (jsight) [inline rules-java/tests/src/test/resources/classmetadatatest/sampleclasses/com/rhc/booking/services/EventServer.java]: Why remove the source code for these?

Comment (mareknovotny) [inline rules-java/tests/src/test/resources/classmetadatatest/sampleclasses/com/rhc/booking/services/EventServer.java]: There is another same source file at src/test/resources/classmetadatatest/src/com/rhc/booking/services/EventServer.java I was confused by duplicated files

**Comment by mareknovotny** [comment]
retest

**Comment by mareknovotny** [comment]
Not sure why this is failing 

```
Input Application:../test-files/Windup1x-javaee-example-tiny.war
Output Path:/var/lib/jenkins/workspace/windup-pr-builder/tests/target/junit2097321704763502794

WARNING: Unrecognized command-line argument: --input
WARNING: Unrecognized command-line argument: ../test-files/Windup1x-javaee-example-tiny.war
WARNING: Unrecognized command-line argument: --output
WARNING: Unrecognized command-line argument: /var/lib/jenkins/workspace/windup-pr-builder/tests/target/junit1193744952824208264
WARNING: Unrecognized command-line argument: --target
WARNING: Unrecognized command-line argument: eap7
WARNING: Unrecognized command-line argument: --includeTags
Windup execution failed due to: null
java.lang.NullPointerException
	at org.jboss.windup.bootstrap.commands.windup.RunWindupCommand.runWindup(RunWindupCommand.java:186)
	at org.jboss.windup.bootstrap.commands.windup.RunWindupCommand.execute(RunWindupCommand.java:73)
	at org.jboss.windup.bootstrap.Bootstrap.executePhase(Bootstrap.java:454)
	at org.jboss.windup.bootstrap.Bootstrap.run(Bootstrap.java:319)
	at org.jboss.windup.bootstrap.Bootstrap.main(Bootstrap.java:100)
	at org.jboss.windup.tests.bootstrap.AbstractBootstrapTest.bootstrap(AbstractBootstrapTest.java:56)
	at org.jboss.windup.tests.bootstrap.migrate.MissingValueOfIncludeTagsTest.missingValueOfIncludeTags(MissingValueOfIncludeTagsTest.java:18)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:367)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:274)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:161)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 14.602 sec <<< FAILURE! - in org.jboss.windup.tests.bootstrap.migrate.MissingValueOfIncludeTagsTest
```

This commit doesn't change the Bootstrap options and locally it passes. The warnings about all unrecognized options are suspicious to me too, only warning about --packages should be logged as I see locally.
@OndraZizka, @jsight Any idea what's happening in windup-pr-builder?

**Comment by jsight** [comment]
retest

**Comment by jsight** [comment]
Yeah, I don't know what was triggering that either. It looks like it is working now, though. Merging...

**Pull request merged**